Patch-Source: https://lists.busybox.net/pipermail/busybox/2024-November/091029.html
---
From 77e364c66bf1208269a257bf23bcdb07ac50b5cd Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Sat, 16 Nov 2024 22:21:12 +0100
Subject: [PATCH] adduser: remove preconfigured GECOS full name field

Pre-configuring "Linux User" as Full Name is confusing a lot of users
downstream.[1] For example, in geary e-mails were sent out as "Linux
User"[2] and I also saw a couple of git commits with this name. This name
does not bring any more improvements than just leaving this field
empty, and let applications most likely fallback to the username.

Alternative would be to set the field to `&` (Capitalized username,
according to passwd(5)) or that we just set the GECOS field downstream.

[1] https://gitlab.postmarketos.org/postmarketOS/pmbootstrap/-/issues/2214
[2] https://gitlab.gnome.org/GNOME/geary/-/issues/1486

Signed-off-by: fossdd <fossdd@pwned.life>
---
 loginutils/adduser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loginutils/adduser.c b/loginutils/adduser.c
index d3c795afa..bfab05203 100644
--- a/loginutils/adduser.c
+++ b/loginutils/adduser.c
@@ -196,7 +196,7 @@ int adduser_main(int argc UNUSED_PARAM, char **argv)
 		bb_simple_error_msg_and_die(bb_msg_perm_denied_are_you_root);
 	}
 
-	pw.pw_gecos = (char *)"Linux User,,,";
+	pw.pw_gecos = (char *)"";
 	/* We assume that newly created users "inherit" root's shell setting */
 	pw.pw_shell = xstrdup(get_shell_name()); /* might come from getpwnam(), need to make a copy */
 	pw.pw_dir = NULL;
-- 
2.47.1

